Library/Homebrew/extend/enumerable.rb

```diff
@@ -0,0 +1,48 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Enumerable
+  # Convert an enumerable to a hash, using the block result as the key and the
+  # element as the value.
+  #
+  #   people.index_by(&:login)
+  #   # => { "nextangle" => <Person ...>, "chade-" => <Person ...>, ...}
+  #
+  #   people.index_by { |person| "#{person.first_name} #{person.last_name}" }
+  #   # => { "Chade- Fowlersburg-e" => <Person ...>, "David Heinemeier Hansson" => <Person ...>, ...}
+  def index_by(&block)
+    if block
+      result = {}
+      each { |elem| result[yield(elem)] = elem }
+      result
+    else
+      T.unsafe(self).to_enum(:index_by) { T.unsafe(self).size if respond_to?(:size) }
```

Comment:
This is really ugly.

Reply:
I'm honestly not entirely sure what's going on here. There's only two usages of `index_by` so if we're wanting to keep it then we don't really need to keep the nil-block case which has a behaviour that's undocumented anyway.

The whole thing can probably be simpified to `each_with_object({}) { |elem, hash| hash[yield(elem)] = elem }`, or replace the two usages with that.

Reply:
I agree that `index_by` can probably be removed entirely and this code inlined given there's only two usages.